### PR TITLE
Guide for setting environment variables in Dagster+ UI

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation
Guide for setting env vars in the Dagster+ UI. Largely pulled from the [existing guide](https://docs.dagster.io/dagster-plus/managing-deployments/environment-variables-and-secrets#using-the-dagster-ui)

## Changelog [New | Bug | Docs]

`NOCHANGELOG`

### CODE REVIEW

**Comment by jamiedemaria** [comment]
* **#24012** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24012?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jamiedemaria and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md [hunk 1]
```diff
@@ -5,4 +5,115 @@ sidebar_position: 1
...
+
+To use the downloaded environment variables for local Dagster development:
+1. Rename the downloaded `env.txt` file to `.env`
+2. Move the file to the directory where you run `dagster dev` or `dagster-webserver`
+3. Run `dagster dev` or `dagster-webserver`
+4. Confirm that the environment variables have been loaded by locating a log message that begins with `Loaded environment variables from .env file`
+
+
+## How to set different values for environment variables in different deployment environments
+You can create multiple instances of the same environment variable key with different values. This allows you to provide different values to different deployment environments. For example, you may want to use different Snowflake credentials for your production deployment than in branch deployments.
```
Comment (jamiedemaria) [inline docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md:107]: this section feels a bit wordy to me. i'm going to revisit it in an hour or so

Comment (PedramNavid) [reply docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md:107]: It honestly doesn't seem to bad to me

File: docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md [hunk 2]
```diff
@@ -5,4 +5,115 @@
...
+
+When you [add an environment variable](#how-to-add-an-environment-variable) you can select the deployment scope and code location scope for the environment variable. You can create multiple environment variables with different values and different scopes to customize the values in different deployment environments.
```
Comment (PedramNavid) [inline docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md:109]: This might be just repeating the previous paragraph though

**Review by PedramNavid — COMMENTED** [review]
Overall looks good! Once the docs are deployed, let me know and I can give it a second look

File: docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md [hunk 3]
```diff
@@ -5,4 +5,115 @@ sidebar_position: 1
...
+2. Click the **Edit** button on the right of the environment variable you want to edit
+3. A modal will open where you can edit the attributes of the environment variable
+4. Click **Save**
+
+## How to delete an environment variable
+To delete an environment variable:
+
+1. [Navigate to the environment variables page](#how-to-find-the-environment-variables-page)
+2. Click the **Trash icon** on the right of the environment variable you want to delete
+3. A modal will open to confirm that you want to delete the environment variable. Click **Yes, delete** to delete the environment variable, or **Cancel** to close the modal and not delete the environment variable.
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md:64]: Similar suggestion here about cutting things down:

```suggestion
## Delete an environment variable

Before you begin, use the deployment switcher to select the right deployment.

You can delete an environment variable by clicking the **trash icon** button in the **Actions** column.
```

File: docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md [hunk 4]
```diff
@@ -5,4 +5,115 @@ sidebar_position: 1
...
+
+## How to view the value an environment variable
+To view the value of an environment variable:
+
+1. [Navigate to the environment variables page](#how-to-find-the-environment-variables-page)
+2. Click the **eye icon** in the **Value** column of the environment variable you want to view
+3. Click the **eye icon** again to hide the value.
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md:71]: And similarly:

```suggestion
## View an environment variable's value

Before you begin, use the deployment switcher to select the right deployment.

You can view the value of an environment variable by clicking the **eye icon** in the **Values** column. Click the icon again to hide the value.
```

File: docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md [hunk 5]
```diff
@@ -5,4 +5,115 @@ sidebar_position: 1
...
+
+:::note
+Viewing an environment variable only reveals the value to you. It doesn't show the value in plaintext to all users. If you navigate away from the environment variables page or reload the page, the value will be hidden again.
+:::
+
+
+
+## How to add an environment variable for a Branch Deployment
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md:79]: Does this need its own section?

File: docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md [hunk 6]
```diff
@@ -5,4 +5,115 @@ sidebar_position: 1
...
+
+
+
+## How to export environment variables locally
+You can export environment variables that have the **Local** scope selected into a `.env` file to use locally.
+
+To download the file:
+1. [Navigate to the environment variables page](#how-to-find-the-environment-variables-page)
+2. Click the **arrow menu** to the right of the **+ Add environment variable** button
+3. Click **Download local environment variables**. A file named `env.txt` will be downloaded
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md:97]: Similarly:

```suggestion
Before you begin, use the deployment switcher to select the right deployment.

1. In the **Environment variables** page, click the **arrow menu** to the right of the **+ Add environment variable** button.
2. Click **Download local environment variables**. 
3. A file named `env.txt` will be downloaded.
```

**Review by erinkcochran87 — COMMENTED** [review]
I'm still thinking on the last section of this guide, but wanted to put up the rest of my comments first. I'll add more in a little bit.

File: docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md (hunk 5 above)
Comment (jamiedemaria) [reply docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md:79]: not really sure! before i wrote anything i started with a list of questions i thought people might have and one of them was "how do i set an env var for a branch deployment?" so i wrote this section. it is completely repetitive, but my thinking was like if you had that specific question you'd be able to go right to this section? happy to remove it though!

File: docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md [hunk 7]
```diff
@@ -5,4 +5,115 @@ sidebar_position: 1
...
+<!-- TODO SCREENSHOT -->
+
+4. Click **Save**
+
+## How to edit an environment variable
+To edit an existing environment variable:
+
+1. [Navigate to the environment variables page](#how-to-find-the-environment-variables-page)
+2. Click the **Edit** button on the right of the environment variable you want to edit
+3. A modal will open where you can edit the attributes of the environment variable
```
[the remark this answers is not in the document]
Comment (jamiedemaria) [reply docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md:56]: i like that! I think i'm going to keep the longer instructions for downloading the .env file because i got tripped up on how to do that and actually load the values when i run dagster locally. otherwise, most of these are really self-explanatory

File: docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md (hunk 5 above)
Comment (jamiedemaria) [reply docs/docs-beta/docs/dagster-plus/deployment/environment-variables/dagster-ui.md:79]: read the section again, and i'm going to delete it

**Comment by jamiedemaria** [comment]
@PedramNavid  docs preview is ready! https://dagster-docs-beta-48r5od1ta-elementl.vercel.app/dagster-plus/deployment/environment-variables/dagster-ui

**Comment by jamiedemaria** [comment]
@erinkcochran87 @PedramNavid or @cmpadden who should the technical reviewer of this be?

**Review by erinkcochran87 — APPROVED** [review]
